tests/test_endpoints.py

```diff
@@ -1101,3 +1111,19 @@ async def test_with_7_x_firmware(
     assert envoy.part_number == part_number
     assert _updater_features(envoy._updaters) == updaters
     assert envoy._supported_features == supported_features
+
+    if supported_features & supported_features.ENPOWER:
+        respx.post(URL_GRID_RELAY).mock(return_value=Response(200, json={}))
+        await envoy.go_on_grid()
+        assert respx.calls.last.request.content == orjson.dumps(
+            {"mains_admin_state": "closed"}
+        )
+        await envoy.go_off_grid()
+        assert respx.calls.last.request.content == orjson.dumps(
+            {"mains_admin_state": "open"}
+        )
+    else:
+        with pytest.raises(EnvoyFeatureNotAvailable):
+            await envoy.go_off_grid()
+        with pytest.raises(EnvoyFeatureNotAvailable):
+            await envoy.go_on_grid()
```

Comment:
I think it would be cleaner to make a new test for the go grid calls so we don't end up with every api call being tested in this function

Reply:
May refactor it in the future, right now there's not a lot else in that test function aside from asserting the snapshot and supported features